### PR TITLE
[Anchor-338] Implement static status callback configuration

### PR DESCRIPTION
**Pull request by JiahuiWho**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

Create client status call back configuration, validate both `custodial` and `non-custodial` type

### Why

[TODO: Why this change is being made. Include any context required to understand the why.]

### Known limitations

Note that for non-custodial client with signing_key provided, we need to check it against TOML file. 

The function `fetchSigningKey` is a copy of `getClientAccountId`(which returns signing_key read from toml file) in `Sep10Service`,  I would love to extract this function to a helper class for reuse.

### CODE REVIEW

File: core/src/main/java/org/stellar/anchor/sep10/Sep10Helper.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import static org.stellar.anchor.util.Log.infoF;
+
+import java.io.IOException;
+import org.stellar.anchor.api.exception.SepException;
+import org.stellar.anchor.util.Sep1Helper;
+import org.stellar.sdk.FormatException;
+import org.stellar.sdk.KeyPair;
+
+public class Sep10Helper {
+  public static String getClientAccountId(String clientDomain) throws SepException {
```
Comment (lijamie98) [inline core/src/main/java/org/stellar/anchor/sep10/Sep10Helper.java:13]: This function is complex. Let's start adding comments.

Comment (lijamie98) [reply core/src/main/java/org/stellar/anchor/sep10/Sep10Helper.java:13]: How about rename to `fetchSigningKey()`?

File: platform/src/main/java/org/stellar/anchor/platform/config/ClientConfig.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+import lombok.Data;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.validation.Errors;
+import org.springframework.validation.ValidationUtils;
+import org.springframework.validation.Validator;
+import org.stellar.anchor.api.exception.SepException;
+import org.stellar.anchor.sep10.Sep10Helper;
+
+@Data
+public class ClientConfig implements Validator {
```
Comment (lijamie98) [inline platform/src/main/java/org/stellar/anchor/platform/config/ClientConfig.java:15]: 💡 Rename to `ClientsConfig` to be consistent with `clients:`

**Review by lijamie98 — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/stellar/anchor/sep10/Sep10Helper.java (hunk 1 above)
Comment (JiahuiWho) [reply core/src/main/java/org/stellar/anchor/sep10/Sep10Helper.java:13]: Yeah this function is moved from Sep10Service for reuse, yeah I agree `fetchSigningKey` would be a more proper name

File: core/src/main/java/org/stellar/anchor/sep10/Sep10Helper.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import static org.stellar.anchor.util.Log.infoF;
+
+import java.io.IOException;
+import org.stellar.anchor.api.exception.SepException;
+import org.stellar.anchor.util.Sep1Helper;
+import org.stellar.sdk.FormatException;
+import org.stellar.sdk.KeyPair;
+
+public class Sep10Helper {
+  public static String fetchSigningKey(String clientDomain) throws SepException {
```
Comment (lijamie98) [inline core/src/main/java/org/stellar/anchor/sep10/Sep10Helper.java:13]: This function is complex. Let's start adding comments.

File: platform/src/main/java/org/stellar/anchor/platform/config/ClientsConfig.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        "signingKey",
+        "empty-client-signing-key",
+        "The client.signingKey cannot be empty and must be defined");
+
+    ValidationUtils.rejectIfEmptyOrWhitespace(
+        errors,
+        "callbackUrl",
+        "empty-client-callback-url",
+        "The client.callbackUrl cannot be empty and must be defined");
+  }
```
Comment (JakeUrban) [inline platform/src/main/java/org/stellar/anchor/platform/config/ClientsConfig.java:53]: Callback URLs should be optional. Some wallets may opt to poll instead, or may use polling first before implementing callbacks at a later stage. Same goes for noncustodial wallets.

Comment (lijamie98) [reply platform/src/main/java/org/stellar/anchor/platform/config/ClientsConfig.java:53]: In that case, please also update the comments of the config YAML file.

**Review by lijamie98 — APPROVED** [review]
LGTM

**Pull request merged**